src/orquestra/sdk/_base/_spaces/_structs.py

```diff
@@ -2,14 +2,15 @@
 # © Copyright 2023 Zapata Computing Inc.
 ################################################################################
 from dataclasses import dataclass
+from typing import Optional
 
 from orquestra.sdk.schema.workflow_run import ProjectId, WorkspaceId
 
 
 @dataclass(frozen=True)
 class ProjectRef:
-    workspace_id: WorkspaceId
-    project_id: ProjectId
+    workspace_id: Optional[WorkspaceId]
+    project_id: Optional[ProjectId]
```

Comment:
**issue**: the original reason why we introduced the `ProjectRef` struct was to encode situations where we need both `workspace_id` and `project_id`, specifically to avoid situations where both can be set to `None` thanks to mypy. This diff goes against the original design. What's the motivation behind doing it?

I haven't dug deeper into it yet, but it seems like instead of making these fields `Optional` we should just change the `foo: ProjectRef` type annotations to `foo: WorkspaceId`.